src/program.ts

```diff
@@ -4743,6 +4768,80 @@ export class Class extends TypedElement {
     }
     return out;
   }
+
+  /** Tests if this class or interface extends the given classs or interface. */
+  extends(other: Class): bool {
+    let base = this.base;
```

Comment:
Shouldn't the method check if `this == other` too? Like so:
```suggestion
    let base = this;
```

Reply:
That would mean that a class extends itself. Probably better to check that manually with an `a == b` where applicable.